### ISSUE TITLE
ExtensionConfigurationException on system properties annotations example provided in the doc

### ISSUE THREAD

**Issue by oliviercailloux** [junit-pioneer/junit-pioneer#473]
I copied the [example](https://junit-pioneer.org/docs/system-properties/) provided in the doc to a [test class in my project](https://github.com/oliviercailloux/JARiS/blob/pioneer/src/test/java/io/github/oliviercailloux/jaris/credentials/MySystemPropertyTest.java). When executing `mvn test`, I get this [exception from pioneer](https://github.com/oliviercailloux/JARiS/runs/2487957466?check_suite_focus=true#step:5:40).

````
[INFO] Running io.github.oliviercailloux.jaris.credentials.MySystemPropertyTest
[ERROR] Tests run: 1, Failures: 0, Errors: 1, Skipped: 0, Time elapsed: 0.006 s <<< FAILURE! - in io.github.oliviercailloux.jaris.credentials.MySystemPropertyTest
[ERROR] test  Time elapsed: 0 s  <<< ERROR!
org.junit.jupiter.api.extension.ExtensionConfigurationException: Don't clear/set the same entry more than once.
Caused by: java.lang.IllegalStateException: Cannot clear and set the following entries at the same time: some property
````

Here are my dependencies.
````
[INFO] --- maven-dependency-plugin:3.1.2:tree (default-cli) @ jaris ---
[INFO] io.github.oliviercailloux:jaris:jar:0.0.10-SNAPSHOT
[INFO] +- org.junit-pioneer:junit-pioneer:jar:1.4.0:test
[INFO] |  +- org.junit.jupiter:junit-jupiter-api:jar:5.7.1:test
[INFO] |  |  \- org.opentest4j:opentest4j:jar:1.2.0:test
[INFO] |  +- org.junit.jupiter:junit-jupiter-params:jar:5.7.1:test
[INFO] |  +- org.junit.platform:junit-platform-commons:jar:1.7.1:test
[INFO] |  \- org.junit.platform:junit-platform-launcher:jar:1.7.1:test
[INFO] +- com.google.jimfs:jimfs:jar:1.2:test
[INFO] +- org.mockito:mockito-core:jar:3.9.0:test
[INFO] |  +- net.bytebuddy:byte-buddy:jar:1.10.20:test
[INFO] |  +- net.bytebuddy:byte-buddy-agent:jar:1.10.20:test
[INFO] |  \- org.objenesis:objenesis:jar:3.2:test
[INFO] +- org.slf4j:slf4j-api:jar:1.7.30:compile
[INFO] +- com.google.guava:guava:jar:30.1.1-jre:compile
[INFO] |  +- com.google.guava:failureaccess:jar:1.0.1:compile
[INFO] |  +- com.google.guava:listenablefuture:jar:9999.0-empty-to-avoid-conflict-with-guava:compile
[INFO] |  +- com.google.code.findbugs:jsr305:jar:3.0.2:compile
[INFO] |  +- org.checkerframework:checker-qual:jar:3.8.0:compile
[INFO] |  +- com.google.errorprone:error_prone_annotations:jar:2.5.1:compile
[INFO] |  \- com.google.j2objc:j2objc-annotations:jar:1.3:compile
[INFO] +- ch.qos.logback:logback-classic:jar:1.2.3:test
[INFO] |  \- ch.qos.logback:logback-core:jar:1.2.3:test
[INFO] \- org.junit.jupiter:junit-jupiter-engine:jar:5.7.1:test
[INFO]    +- org.apiguardian:apiguardian-api:jar:1.1.0:test
[INFO]    \- org.junit.platform:junit-platform-engine:jar:1.7.1:test
````

I must be missing something stupid, but I can’t find out what.

**Comment by Michael1993** [comment]
It's very likely that this is actually a bug related to #460. Please give us some time while we look into it.

Pinging @beatngu13, just in case.

**Comment by beatngu13** [comment]
@oliviercailloux thx for reporting, I can confirm that this test fails.

@Michael1993 not sure if related, all I can say so far is that [this (nested) test](https://github.com/junit-pioneer/junit-pioneer/blob/main/src/test/java/org/junitpioneer/jupiter/SystemPropertyExtensionTests.java#L143) obviously doesn't test what we want it to test.

I will further investigate this issue.

**Comment by beatngu13** [comment]
Update: [Our test](https://github.com/junit-pioneer/junit-pioneer/blob/main/src/test/java/org/junitpioneer/jupiter/SystemPropertyExtensionTests.java#L143) passes because it's like adapting Olivier's test as follows:

```java
@ClearSystemProperty(key = "some property")
class MySystemPropertyTest {

	@Test
	@ClearSystemProperty(key = "another property")
	@SetSystemProperty(key = "some property", value = "new value")
	void test() {
		assertEquals("new value", System.getProperty("some property"));
	}
}
```

* Before all callback:
  * Entries to clear: `some property`
  * Entries to set: n/a
* Before each callback:
  * Entries to clear: `another property`
  * Entries to set: `some property`

When `another property` is not cleared, the annotation lookup `PioneerAnnotationUtils#findClosestEnclosingRepeatableAnnotations(ExtensionContext, Class<A)` finds `some property` both for entries to clear as well as entries to set. That is, clearing `another property` somewhat "shadows" the class-level annotation as the underlying search stops on "closest".

So, I think the issue is unrelated to #460. This would be a problem even if the extension(s) would only implemented `BeforeEachCallback`/`AfterEachCallback`. Nevertheless, it's a bug.

**Comment by nipafx** [comment]
Thank you @oliviercailloux for reporting the bug and giving such a clear explanation. That made it easy to reproduce.

@beatngu13 fixed it and we just released v1.4.1 with the fix. Have fun!

**Comment by oliviercailloux** [comment]
It is a pleasure to contribute (very modestly) to such an efficient workflow as yours.
